src/crimson/osd/osd_operation_sequencer.h

```diff
@@ -69,10 +69,40 @@ class OpSequencer {
   const ClientRequest* get_last_issued() const {
     return last_issued;
   }
-  void finish_op(const ClientRequest* const op) {
+  void finish_op_in_order(const ClientRequest* const op) {
     assert(op->get_id() > last_completed->get_id());
     last_completed = op;
   }
+  void finish_op_out_of_order(ClientRequest* const this_op) {
+    assert(this_op);
+    for (auto it = OperationRegistryI::op_list::s_iterator_to(*this_op);
+         it != OperationRegistryI::op_list::s_iterator_to(*last_issued);
+         ++it) {
+      // we're iterating over the operation registry of all ClientRequests.
+      // this list aggrates every single instance in the system, and thus
+      // we need to skip operations coming from different client's session
+      // or targeting different PG. The op we're interested in should be
```

Comment:
I think the op sequencer is already unique for each connection/pg pair, which, I think, makes the op sequencer allow for out-of-order op execution from different connection/pg pair. Am I missing anything?